examples/dnn_introduction3_ex.cpp

```diff
@@ -0,0 +1,180 @@
+    {
+        l.set_learning_rate_multiplier(new_learning_rate_multiplier);
+        l.set_bias_learning_rate_multiplier(new_learning_rate_multiplier);
+    }
+
+    // fully connected layers (with and without bias)
+    template<unsigned long num_outputs, fc_bias_mode bias_mode>
+    void set_learning_rate_multiplier(fc_<num_outputs, bias_mode>& l) const
+    {
+        l.set_learning_rate_multiplier(new_learning_rate_multiplier);
+        if (bias_mode == FC_HAS_BIAS)
+            l.set_bias_learning_rate_multiplier(new_learning_rate_multiplier);
+    }
+
+    template<typename input_layer_type>
+    void operator()(size_t , input_layer_type& )  const
+    {
+        // ignore other layers
+    }
+
+    template <typename T, typename U, typename E>
+    void operator()(size_t , add_layer<T,U,E>& l)  const
+    {
+        set_learning_rate_multiplier(l.layer_details());
+    }
+
+private:
+
+    double new_learning_rate_multiplier;
+};
+
+
+int main() try
+{
+    // Let's instantiate our network in train mode
+    model::train net;
+
+    // We create a new scope so that resources from the loaded network are freed
+    // automatically when leaving the scope.
+    {
+        // Now, let's define the classic ResNet50 network and load the pretrained model on ImageNet
+        resnet<bn_con>::l50 resnet50;
+        std::vector<string> labels;
+        deserialize("resnet50_1000_imagenet_classifier.dnn") >> resnet50 >> labels;
+        // For transfer learning, we are only interested in the ResNet50's backbone, which lays
+        // below the loss and the fc layers, so we can extract it as:
+        auto backbone = std::move(resnet50.subnet().subnet());
+        // We can now assign ResNet50's backbone to our network skipping the different layers,
+        // in our case, the loss layer and the fc layer:
+        net.subnet().subnet() = backbone;
+
+        // An alternative way to use the pretrained network on a different network is
+        // to extract the relevant part of the network (we remove loss and fc layers),
+        // stack the new layers on top of it and assign the network.
+        using net_type = loss_metric<fc_no_bias<128, decltype(backbone)>>;
+        net_type net2;
+        net2.subnet().subnet() = backbone;
+    }
+    cin.get();
+
+    // We can use the visit_layers function to modify the learning rate of the whole network:
+    visit_layers(net, visitor_learning_rate_multiplier(0.01));
+
+    // Usually, we want to freeze the network, except for the top layers:
+    visit_layers(net.subnet().subnet(), visitor_learning_rate_multiplier(0));
+
+    // However, sometimes we might want to adjust the learning rate differently thoughout the
+    // network.  Here we show how to use the visit_layers_range to adjust the learning rate
+    // different at the different ResNet50's convolutional blocks:
+    visit_layers_range<  0,   2, decltype(net), visitor_learning_rate_multiplier>(net, visitor_learning_rate_multiplier(1));
```

Comment:
Does `visit_layers_range<0,2>(net, visitor_learning_rate_multiplier(1));` not compile?  It should.  Something is funky if it's not.

Reply:
Oh, yes, it does compile!